### PR TITLE
heptabase: init at 1.35.3

### LINKED ISSUE #313853: Package request: heptabase
**Project description**

**Metadata**

* homepage URL: https://heptabase.com/download
* source URL: none
* license: unfree
* platforms: unix, linux, darwin

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by luftmensch-luftmensch**
## Description of changes
Closes  #313853

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by luftmensch-luftmensch** [comment]
```
> nix build .#heptabase
> tree result/
 result
├──  bin
│  └──  heptabase -> /nix/store/7d3lag5j5ipj3z8ifp6y3k2441x58lxp-heptabase-1.32.13-bwrap
└──  share
   ├──  applications
   │  └──  project-meta.desktop
   └──  icons
      └──  hicolor
         └──  0x0
            └──  apps
               └──  project-meta.png
```

**Review by MaxwellDupre — APPROVED** [review]
LGTM

**Comment by luftmensch-luftmensch** [comment]
@MaxwellDupre since the PR is still opened I've bumped the version derivation to the latest available

+ [Release notes](https://github.com/heptameta/project-meta/releases/tag/v1.32.15)

**Comment by pbsds** [comment]
```
┃        > trying https://github.com/heptameta/project-meta/releases/download/v1.32.16/Heptabase-1.32.16.AppImage
┃        >   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
┃        >                                  Dload  Upload   Total   Spent    Left  Speed
┃        >   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
┃        > curl: (22) The requested URL returned error: 404
┃        > error: cannot download Heptabase-1.32.16.AppImage from any mirror
```

**Comment by luftmensch-luftmensch** [comment]
> ```
> ┃        > trying https://github.com/heptameta/project-meta/releases/download/v1.32.16/Heptabase-1.32.16.AppImage
> ┃        >   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
> ┃        >                                  Dload  Upload   Total   Spent    Left  Speed
> ┃        >   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
> ┃        > curl: (22) The requested URL returned error: 404
> ┃        > error: cannot download Heptabase-1.32.16.AppImage from any mirror
> ```

Apparently the changed something in the meantime. I've bumped the derivation to use the latest version available

File: pkgs/by-name/he/heptabase/package.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    hash = "sha256-0wPtAvBuc8J+XJqZXDzV+RL/37NPZjKcs3do1jQkE14=";
+  };
+
+  appimageContents = appimageTools.extractType2 { inherit pname version src; };
+in
+appimageTools.wrapType2 {
+  inherit pname version src;
+
+  extraInstallCommands = ''
+    install -Dm444 ${appimageContents}/project-meta.desktop -t $out/share/applications
```
Comment (pbsds) [inline pkgs/by-name/he/heptabase/package.nix:20]:
```suggestion
    install -Dm444 ${appimageContents}/project-meta.desktop -t $out/share/applications/heptabase.desktop
```

Comment (pbsds) [reply pkgs/by-name/he/heptabase/package.nix:20]: the `Exec=` field in this file is:
```
Exec=AppRun --no-sandbox %U
```

Please patch it to
```
Exec=heptabase %U
```

File: pkgs/by-name/he/heptabase/package.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    cp -r ${appimageContents}/usr/share/icons $out/share
```
Comment (pbsds) [inline pkgs/by-name/he/heptabase/package.nix:21]: The icon ends up at `results/heptabase/share/icons/hicolor/0x0/apps/project-meta.png` which has two errors:

* `project-meta` should be `heptabase`
* `0x0` should be `512x512`

Please address both and update the `Icon=` field in the desktop entry to reflect the change

**Review by pbsds — COMMENTED** [review]
Please squash the commits and reword it to reflect the current version

**Comment by luftmensch-luftmensch** [comment]
@pbsds I've applied all the requested changes and bumped version to [1.35.3](https://github.com/heptameta/project-meta/releases/tag/v1.35.3). Please let me know if there are other changes to do

**Comment by pbsds** [comment]
image
